prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -170,7 +167,7 @@ class FHIRRouter(
                 // TODO: merge with mapped condition filter (see https://github.com/CDCgov/prime-reportstream/issues/12705)
                 // If the receiver does not have a condition filter set send the entire bundle to the translate step
                 var receiverBundle = if (receiver.conditionFilter.isEmpty()) {
-                    bundle
+                    bundle.copy()
```

Comment:
+++

Reply:
The bundle must get copied before any modification to prevent filters bleeding across receivers. But looking at it again, this should move to the code block that evaluates the `mappedConditionFilter` (and only happen if that filter will run).

I'll make the update.